mantle/platform/metal.go

```diff
@@ -0,0 +1,461 @@
+		}
+	} else {
+		if inst.CosaBuild.BuildArtifacts.LiveKernel.Path == "" {
+			return nil, fmt.Errorf("build %s has no live installer kernel", inst.CosaBuild.Name)
+		}
+		mach, err = inst.runLive(&kernelSetup{
+			kernel:    inst.CosaBuild.BuildArtifacts.LiveKernel.Path,
+			initramfs: inst.CosaBuild.BuildArtifacts.LiveInitramfs.Path,
+		})
+		if err != nil {
+			return nil, errors.Wrapf(err, "testing live installer")
+		}
+	}
+
+	return mach, nil
+}
+
+func (inst *InstalledMachine) Destroy() error {
+	if inst.tempdir != "" {
+		return os.RemoveAll(inst.tempdir)
+	}
+	return nil
+}
+
+type kernelSetup struct {
+	kernel, initramfs string
+}
+
+type pxeSetup struct {
+	tftpipaddr    string
+	boottype      string
+	networkdevice string
+	bootindex     string
+	pxeimagepath  string
+
+	// bootfile is initialized later
+	bootfile string
+}
+
+type installerRun struct {
+	inst    *Install
+	builder *QemuBuilder
+
+	builddir string
+	tempdir  string
+	tftpdir  string
+
+	metalimg  string
+	metalname string
+
+	baseurl string
+
+	kern kernelSetup
+	pxe  pxeSetup
+}
+
+func absSymlink(src, dest string) error {
+	src, err := filepath.Abs(src)
+	if err != nil {
+		return err
+	}
+	return os.Symlink(src, dest)
+}
+
+func (inst *Install) setupTftpDir(builddir, tftpdir, metalimg string, kern *kernelSetup) error {
+	serializedConfig := []byte(inst.ignition)
+	if err := ioutil.WriteFile(filepath.Join(tftpdir, "config.ign"), serializedConfig, 0644); err != nil {
+		return err
+	}
+
+	metalIsCompressed := !strings.HasSuffix(metalimg, ".raw")
+	metalname := metalimg
+	if !metalIsCompressed {
+		fmt.Println("Compressing metal image")
+		metalimgpath := filepath.Join(builddir, metalimg)
+		srcf, err := os.Open(metalimgpath)
+		if err != nil {
+			return err
+		}
+		defer srcf.Close()
+		metalname = metalname + ".gz"
+		destf, err := os.OpenFile(filepath.Join(tftpdir, metalname), os.O_RDWR|os.O_CREATE, 0755)
+		if err != nil {
+			return err
+		}
+		defer destf.Close()
+		cmd := exec.Command("gzip", "-1")
+		cmd.Stdin = srcf
+		cmd.Stdout = destf
+		if err := cmd.Run(); err != nil {
+			return errors.Wrapf(err, "running gzip")
+		}
+	} else {
+		if err := absSymlink(filepath.Join(builddir, metalimg), filepath.Join(tftpdir, metalimg)); err != nil {
+			return err
+		}
+	}
+
+	for _, name := range []string{kern.kernel, kern.initramfs} {
+		if err := absSymlink(filepath.Join(builddir, name), filepath.Join(tftpdir, name)); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (inst *Install) setup(kern *kernelSetup) (*installerRun, error) {
+	if kern.kernel == "" {
+		return nil, fmt.Errorf("Missing kernel artifact")
+	}
+	if kern.initramfs == "" {
+		return nil, fmt.Errorf("Missing initramfs artifact")
+	}
+
+	builder := NewBuilder(inst.Board, "", false)
+	builder.Firmware = inst.Firmware
+	builder.AddDisk(&Disk{
+		Size: "12G", // Arbitrary
+	})
+
+	// This applies just in the legacy case
+	builder.Memory = 1542
```

Comment:
@cgwalters a request - can this be changed to 1536 like before . On ppc64le this fails with: https://bugzilla.redhat.com/show_bug.cgi?id=1249006 . The ppc64le qemu folks put a fix in libvirt to round it up, but QEMU still wants it be aligned to 256M for ppc64le

Reply:
Yep, fixed thanks!